msm/src/serialization/witness.rs

```diff
@@ -15,6 +16,10 @@ pub struct Env<Fp> {
     /// Used for the decomposition in base 4 of the last limb of the foreign
     /// field Kimchi gate
     pub intermediate_limbs: [Fp; N_INTERMEDIATE_LIMBS],
+
+    // Boxing to avoid stack overflow
```

Comment:
Just curious: what's the problem and how boxing solves it? I thought `Box` is just a pointer?

Reply:
It is allocating on the heap the vector, and use a pointer in the stack to this heap memory allocation

Reply:
But does it not /always/ do that with vectors? 

(ah, I guess with vectors but not with slices?)